test/include/fleet_provisioning_config.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * AWS IoT Fleet Provisioning Client v1.0.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file fleet_provisioning_config.h
+ * @brief Config values for testing the AWS IoT Fleet Provisioning Client Library.
+ */
+
+#ifndef FLEET_PROVISIONING_CONFIG_H_
+#define FLEET_PROVISIONING_CONFIG_H_
+
+#include <stdio.h>
```

Comment:
You can refer to the SNTP configuration for enabling logging with the warning level prefixes
https://github.com/FreeRTOS/coreSNTP/blob/main/test/unit-test/core_sntp_config.h#L35-L54

Reply:
The SNTP one seems wrong as it would only work with LOGGING_LEVEL_DEBUG, since other levels dont define all the macros, and thus would fail where, for example, LogDebug is used.
Additionally this is just for the tests, where we want all the log macro invocations to be tested so that incorrect parameters do not escape scrutiny. I don't see value in adding the option to only test some levels, which will never be used.

Reply:
>The SNTP one seems wrong as it would only work with LOGGING_LEVEL_DEBUG,

With `LOGGING_LEVEL_DEBUG`, all the logging macros are enabled as you can see here: https://github.com/FreeRTOS/coreSNTP/blob/main/test/unit-test/core_sntp_config.h#L49-L53

Reply:
Additionally the SNTP one uses variadic macros which are a c99 feature.

Reply:
Yes, but it will not work with LOGGING_LEVEL_ERROR or the others set instead as it does not define all the macros.

Reply:
One of the minor reasons for providing the different logging level control is to show an alternative definition of the logging macros with printf. Irrespective of whether the logging level control is provided, it is valuable to add the logging level prefixes

Reply:
That is fair, I will add the prefixes

Reply:
>Yes, but it will not work with LOGGING_LEVEL_ERROR or the others set instead as it does not define all the macros.

We depend on the `<library>_config_defaults.h` file for the default configurations of the macros.